.github/actions/setup-node/action.yml

```diff
@@ -0,0 +1,33 @@
+name: 'setup'
+description: 'composite for shared job setup'
+inputs:
+  node-version:
+    description: 'node version'
+    default: '24'
+  install-deps:
+    description: 'install deps'
+    default: 'true'
+
+runs:
+  using: composite
+  steps:
```

Comment:
these steps are required in all of our jobs, this makes it easier to update and reduced code duplication